packages/web-components/components/document/src/document-capture/DocumentCapture.js

```diff
@@ -339,71 +339,82 @@ class DocumentCapture extends HTMLElement {
-      this.shadowRoot.querySelector('.actions').hidden = false;
-      if (!videoExists) {
-        videoContainer.prepend(canvas);
+      video.autoplay = true;
+      video.playsInline = true;
+      if ('srcObject' in video) {
+        video.srcObject = stream;
+      } else {
+        video.src = window.URL.createObjectURL(stream);
       }
-    };
 
-    const onVideoStart = () => {
-      if (video.paused || video.ended) return;
-      video.removeEventListener('playing', onVideoStart);
-      const aspectRatio = video.videoWidth / video.videoHeight;
-      const portrait = aspectRatio < 1;
+      canvas.width = videoContainer.clientWidth;
+      canvas.height = (videoContainer.clientWidth * 9) / 16;
       if (this.isPortraitCaptureView) {
-        this.updatePortraitId(canvas, video);
-        requestAnimationFrame(onVideoStart);
-        return;
+        canvas.height = (videoContainer.clientWidth * 16) / 9;
       }
 
-      if (portrait) {
-        videoContainer.classList.add('mobile-camera-screen');
-        const intermediateCanvas = document.createElement('canvas');
-        this._capturePortraitToLandscapeImage(intermediateCanvas, video);
-        this._drawLandscapeImageFromCanvas(canvas, intermediateCanvas);
-      } else {
-        this._drawLandscapeImage(canvas, video);
-      }
-      requestAnimationFrame(onVideoStart);
-    };
+      video.onloadedmetadata = () => {
+        video.play();
+
+        this.shadowRoot.querySelector('#loader').hidden = true;
+        this.shadowRoot.querySelector('.id-video').hidden = false;
+        this.shadowRoot.querySelector('.actions').hidden = false;
+        if (!videoExists) {
+          videoContainer.prepend(canvas);
+        }
+      };
+
+      const onVideoStart = () => {
+        if (video.paused || video.ended) return;
+        video.removeEventListener('playing', onVideoStart);
+        const aspectRatio = video.videoWidth / video.videoHeight;
+        const portrait = aspectRatio < 1;
+        if (this.isPortraitCaptureView) {
+          this.updatePortraitId(canvas, video);
+          requestAnimationFrame(onVideoStart);
+          return;
+        }
+
+        if (portrait) {
+          videoContainer.classList.add('mobile-camera-screen');
+          const intermediateCanvas = document.createElement('canvas');
+          this._capturePortraitToLandscapeImage(intermediateCanvas, video);
+          this._drawLandscapeImageFromCanvas(canvas, intermediateCanvas);
+        } else {
+          this._drawLandscapeImage(canvas, video);
+        }
+        requestAnimationFrame(onVideoStart);
+      };
 
-    video.addEventListener('playing', onVideoStart);
+      video.addEventListener('playing', onVideoStart);
 
-    this._IDStream = stream;
-    this._IDVideo = video;
+      this._IDStream = stream;
+      this._IDVideo = video;
+    } catch (error) {
+      if (error.name !== 'AbortError') {
+        console.error(error);
```

Comment:
why just log the error in the components? is there no error callback in the components?

Reply:
Done